### ISSUE TITLE
Teku fails to start due to "fatal error in Eth1DepositManager"

### ISSUE THREAD

**Issue by kdelwat** [Consensys-Incorporated/teku#2980]
### Description

I'm trying to run a validator on the Zinken testnet. When I try to start Teku, I get a fatal error:

```
FATAL - Exiting due to fatal error in Eth1DepositManager java.util.concurrent.CompletionException: tech.pegasys.teku.service.serviceutils.FatalServiceFailureException: java.util.concurrent.CompletionException: tech.pegasys.teku.pow.exception.InvalidDepositEventsException: Expected next deposit at index 162, but got 187 (See log file for full stack trace)
```

This happens on start up every time. The full stack trace from the logs is:

```
2020-10-14 23:59:59.341+00:00 | pool-16-thread-572 | FATAL | teku-status-log | PLEASE CHECK YOUR ETH1 NODE | Encountered a problem retrieving deposit events from eth1 endpoint.
java.util.concurrent.CompletionException: java.lang.RuntimeException: java.util.concurrent.CompletionException: tech.pegasys.teku.pow.exception.InvalidDepositEventsException: Expected next deposit at index 162, but got 187
	at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:314) ~[?:?]
	at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:319) ~[?:?]
	at java.util.concurrent.CompletableFuture.uniHandle(CompletableFuture.java:932) ~[?:?]
	at java.util.concurrent.CompletableFuture$UniHandle.tryFire(CompletableFuture.java:907) ~[?:?]
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506) ~[?:?]
	at java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2073) ~[?:?]
	at tech.pegasys.teku.infrastructure.async.SafeFuture.lambda$propagateResult$2(SafeFuture.java:132) ~[teku-infrastructure-async-0.12.9-SNAPSHOT.jar:0.12.9-dev-1ba97581]
	at java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:859) ~[?:?]
	at java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:837) ~[?:?]
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506) ~[?:?]
	at java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2073) ~[?:?]
	at tech.pegasys.teku.infrastructure.async.SafeFuture.lambda$propagateResult$2(SafeFuture.java:132) ~[teku-infrastructure-async-0.12.9-SNAPSHOT.jar:0.12.9-dev-1ba97581]
	at java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:859) ~[?:?]
	at java.util.concurrent.CompletableFuture.uniWhenCompleteStage(CompletableFuture.java:883) ~[?:?]
	at java.util.concurrent.CompletableFuture.whenComplete(CompletableFuture.java:2251) ~[?:?]
	at tech.pegasys.teku.infrastructure.async.SafeFuture.whenComplete(SafeFuture.java:466) ~[teku-infrastructure-async-0.12.9-SNAPSHOT.jar:0.12.9-dev-1ba97581]
	at tech.pegasys.teku.infrastructure.async.SafeFuture.whenComplete(SafeFuture.java:28) ~[teku-infrastructure-async-0.12.9-SNAPSHOT.jar:0.12.9-dev-1ba97581]
	at tech.pegasys.teku.infrastructure.async.SafeFuture.propagateResult(SafeFuture.java:127) ~[teku-infrastructure-async-0.12.9-SNAPSHOT.jar:0.12.9-dev-1ba97581]
	at tech.pegasys.teku.infrastructure.async.SafeFuture.lambda$exceptionallyCompose$23(SafeFuture.java:319) ~[teku-infrastructure-async-0.12.9-SNAPSHOT.jar:0.12.9-dev-1ba97581]
	at java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:859) ~[?:?]
	at java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:837) ~[?:?]
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506) ~[?:?]
	at java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2073) ~[?:?]
	at tech.pegasys.teku.infrastructure.async.SafeFuture.lambda$propagateResult$2(SafeFuture.java:132) ~[teku-infrastructure-async-0.12.9-SNAPSHOT.jar:0.12.9-dev-1ba97581]
	at java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:859) ~[?:?]
	at java.util.concurrent.CompletableFuture.uniWhenCompleteStage(CompletableFuture.java:883) ~[?:?]
	at java.util.concurrent.CompletableFuture.whenComplete(CompletableFuture.java:2251) ~[?:?]
	at tech.pegasys.teku.infrastructure.async.SafeFuture.whenComplete(SafeFuture.java:466) ~[teku-infrastructure-async-0.12.9-SNAPSHOT.jar:0.12.9-dev-1ba97581]
	at tech.pegasys.teku.infrastructure.async.SafeFuture.whenComplete(SafeFuture.java:28) ~[teku-infrastructure-async-0.12.9-SNAPSHOT.jar:0.12.9-dev-1ba97581]
	at tech.pegasys.teku.infrastructure.async.SafeFuture.propagateResult(SafeFuture.java:127) ~[teku-infrastructure-async-0.12.9-SNAPSHOT.jar:0.12.9-dev-1ba97581]
	at tech.pegasys.teku.infrastructure.async.SafeFuture.lambda$exceptionallyCompose$23(SafeFuture.java:319) ~[teku-infrastructure-async-0.12.9-SNAPSHOT.jar:0.12.9-dev-1ba97581]
	at java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:859) ~[?:?]
	at java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:837) ~[?:?]
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506) ~[?:?]
	at java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2073) ~[?:?]
	at tech.pegasys.teku.infrastructure.async.SafeFuture.lambda$propagateResult$2(SafeFuture.java:132) ~[teku-infrastructure-async-0.12.9-SNAPSHOT.jar:0.12.9-dev-1ba97581]
	at java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:859) ~[?:?]
	at java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:837) ~[?:?]
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506) ~[?:?]
	at java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2073) ~[?:?]
	at org.web3j.utils.Async.lambda$run$1(Async.java:38) ~[core-4.6.2.jar:?]
	at java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1736) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
Caused by: java.lang.RuntimeException: java.util.concurrent.CompletionException: tech.pegasys.teku.pow.exception.InvalidDepositEventsException: Expected next deposit at index 162, but got 187
	at tech.pegasys.teku.pow.DepositProcessingController.onSubscriptionDepositRequestFailed(DepositProcessingController.java:178) ~[teku-pow-0.12.9-SNAPSHOT.jar:0.12.9-dev-1ba97581]
	at tech.pegasys.teku.pow.DepositProcessingController.onSubscriptionDepositRequestFailed(DepositProcessingController.java:170) ~[teku-pow-0.12.9-SNAPSHOT.jar:0.12.9-dev-1ba97581]
	at tech.pegasys.teku.pow.DepositProcessingController.lambda$fetchLatestDepositsOneBlockAtATime$5(DepositProcessingController.java:154) ~[teku-pow-0.12.9-SNAPSHOT.jar:0.12.9-dev-1ba97581]
	at tech.pegasys.teku.infrastructure.async.SafeFuture.lambda$finish$18(SafeFuture.java:256) ~[teku-infrastructure-async-0.12.9-SNAPSHOT.jar:0.12.9-dev-1ba97581]
	at java.util.concurrent.CompletableFuture.uniHandle(CompletableFuture.java:930) ~[?:?]
	... 42 more
Caused by: java.util.concurrent.CompletionException: tech.pegasys.teku.pow.exception.InvalidDepositEventsException: Expected next deposit at index 162, but got 187
	at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:314) ~[?:?]
	at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:319) ~[?:?]
	at java.util.concurrent.CompletableFuture$UniCompose.tryFire(CompletableFuture.java:1081) ~[?:?]
	... 41 more
Caused by: tech.pegasys.teku.pow.exception.InvalidDepositEventsException: Expected next deposit at index 162, but got 187
	at tech.pegasys.teku.pow.exception.InvalidDepositEventsException.expectedDepositAtIndex(InvalidDepositEventsException.java:26) ~[teku-pow-0.12.9-SNAPSHOT.jar:0.12.9-dev-1ba97581]
	at tech.pegasys.teku.pow.ValidatingEth1EventsPublisher.assertDepositEventIsValid(ValidatingEth1EventsPublisher.java:56) ~[teku-pow-0.12.9-SNAPSHOT.jar:0.12.9-dev-1ba97581]
	at tech.pegasys.teku.pow.ValidatingEth1EventsPublisher.onDepositsFromBlock(ValidatingEth1EventsPublisher.java:46) ~[teku-pow-0.12.9-SNAPSHOT.jar:0.12.9-dev-1ba97581]
	at tech.pegasys.teku.pow.DepositFetcher.postDeposits(DepositFetcher.java:189) ~[teku-pow-0.12.9-SNAPSHOT.jar:0.12.9-dev-1ba97581]
	at tech.pegasys.teku.pow.DepositFetcher.postEventsForBlock(DepositFetcher.java:155) ~[teku-pow-0.12.9-SNAPSHOT.jar:0.12.9-dev-1ba97581]
	at tech.pegasys.teku.pow.DepositFetcher.postDepositEvents(DepositFetcher.java:126) ~[teku-pow-0.12.9-SNAPSHOT.jar:0.12.9-dev-1ba97581]
	at tech.pegasys.teku.pow.DepositFetcher.lambda$postDepositEvents$3(DepositFetcher.java:142) ~[teku-pow-0.12.9-SNAPSHOT.jar:0.12.9-dev-1ba97581]
	at java.util.concurrent.CompletableFuture$UniCompose.tryFire(CompletableFuture.java:1072) ~[?:?]
	... 41 more
```

There doesn't seem to be anything amiss with the Eth 1.0 node, the logs just show block imports:

```
...
Oct 15 04:36:25 ip-172-31-26-186 besu[97792]: 2020-10-15 04:36:25.254+00:00 | EthScheduler-Workers-1 | INFO  | PersistBlockTask | Imported #3,577,625 / 4 tx / 0 om / 110,575 (1.4%) gas / (0xc12e4bb5f6043beb0f35a06ac0a161d98e7b506e46d02c07c1a02ff4a31854aa) in 0.023s. Peers: 3
Oct 15 04:36:25 ip-172-31-26-186 besu[97792]: 2020-10-15 04:36:25.721+00:00 | EthScheduler-Workers-0 | INFO  | PersistBlockTask | Imported #3,577,625 / 3 tx / 0 om / 89,571 (1.1%) gas / (0xe7a5abe02c5089d07e856a5427fcb82c858113108271d24cd81716ab4a079682) in 0.047s. Peers: 3
Oct 15 04:36:40 ip-172-31-26-186 besu[97792]: 2020-10-15 04:36:40.695+00:00 | EthScheduler-Workers-0 | INFO  | PersistBlockTask | Imported #3,577,626 / 2 tx / 0 om / 44,092 (0.6%) gas / (0xa46421f324669bffa533dbdfad3efe110f43d4a339c9d498061fc9eb46213d37) in 0.007s. Peers: 3
Oct 15 04:36:42 ip-172-31-26-186 besu[97792]: 2020-10-15 04:36:42.463+00:00 | EthScheduler-Workers-1 | INFO  | PersistBlockTask | Imported #3,577,626 / 2 tx / 0 om / 70,675 (0.9%) gas / (0x1fc02b7f4e43d1ca1d64d311c89ba0437525f654e472172e580faaaad97270f1) in 0.059s. Peers: 3
...
```

### Steps to Reproduce

* Run Teku with: 

```
teku \
    --validators-key-files /home/ubuntu/validator_keys/keystore-m_12381_3600_0_0_0-1602024580.json \
    --validators-key-password-files /home/ubuntu/validator_keys/password.txt \
    --network=zinken \
    --metrics-enabled \
    --rest-api-enabled=true \
    --rest-api-docs-enabled=true \
    --eth1-endpoint=http://localhost:8545
```

* For the Eth1 node, I'm using Besu 1.5.0 without any custom configuration:

```
/home/ubuntu/besu-1.5.0/bin/besu --network=goerli --rpc-http-enabled
```

### Versions
* Software version: `0.12.9-dev-1ba97581`
* Besu version: `1.5.0`
* Java version: `openjdk version "11.0.8" 2020-07-14`
* OS Name & Version: Ubuntu 20.04.1 LTS
* Kernel Version: `Linux ip-172-31-26-186 5.4.0-1024-aws #24-Ubuntu SMP Sat Sep 5 06:19:55 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux`
* Cloud VM AWS EC2 t3.small

### Additional Information

Please let me know what other information would be useful to start debugging this!

**Comment by ajsutton** [comment]
@kdelwat This looks like you've run into https://github.com/hyperledger/besu/issues/1153 where Besu is failing to return some of the logs from the deposit contract.  It was fixed in Besu 1.5.5.  It will take a 20-30 mins to rebuild its indices after you upgrade Besu and you'll likely then need to restart Teku but it should make progress after that.

**Comment by kdelwat** [comment]
@ajsutton That seems to have fixed it, thank you :smile:
